In [1]:
import os
print(os.getcwd())

C:\Users\ehman\Seoul_Land_Admission


In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

!pip install xlrd

import xlrd

!pip install openpyxl

import openpyxl

import datetime

You should consider upgrading via the 'c:\users\ehman\anaconda3\python.exe -m pip install --upgrade pip' command.


You should consider upgrading via the 'c:\users\ehman\anaconda3\python.exe -m pip install --upgrade pip' command.


## 2018년 데이터 로딩

In [3]:
seoulland_admission_201801 = pd.read_excel('Data/서울대공원 입장객 정보_2018년.xls',header=[1,3],sheet_name=0)

In [4]:
seoulland_admission_201812 = pd.read_excel('Data/서울대공원 입장객 정보_2018년.xls',header=[1,3],sheet_name=11)

In [5]:
seoulland_admission_201801.head()

구분           월별입장객                 동,식물원            ...  \
                    날짜 요일     날씨  유료합계 무료합계    총계 외국인계  유료소계    어른 청소년  ...   
0  2018-01-01 00:00:00  월     맑음  1688  628  2316   40  1683  1375  99  ...   
1  2018-01-02 00:00:00  화  구름 조금   403  585   988   22   401   342  15  ...   
2  2018-01-03 00:00:00  수  구름 조금   192  502   694    4   191   153  22  ...   
3  2018-01-04 00:00:00  목     흐림   165  506   671   14   163   132   9  ...   
4  2018-01-05 00:00:00  금  구름 조금   189  569   758    8   188   145  12  ...   

             테마가든                               
  무료소계    총계 유료소계 어른 청소년 어린이 외국인 단체입장 무료소계  총계  
0  619  2302    5  5   0   0   0    0    9  14  
1  578   979    2  2   0   0   0    0    7   9  
2  502   693    1  0   0   1   0    0    0   1  
3  488   651    2  2   0   0   0    0   18  20  
4  567   755    1  1   0   0   0    0    2   3  

[5 rows x 23 columns]

In [6]:
seoulland_admission_201812.head()

구분           월별입장객                  동,식물원            ...  \
                    날짜 요일     날씨  유료합계  무료합계    총계 외국인계  유료소계    어른 청소년  ...   
0  2018-12-01 00:00:00  토     맑음  2440  1565  4005   46  2375  2046  35  ...   
1  2018-12-02 00:00:00  일     흐림  2221  1160  3381   26  2166  1800  80  ...   
2  2018-12-03 00:00:00  월     흐림    55   301   356    0    55    44  10  ...   
3  2018-12-04 00:00:00  화  구름 많음   144   466   610    4   139   111  23  ...   
4  2018-12-05 00:00:00  수  구름 많음   213   602   815   19   213   150   7  ...   

              테마가든                                 
   무료소계    총계 유료소계  어른 청소년 어린이 외국인 단체입장 무료소계   총계  
0  1520  3895   65  60   0   5   0    0   45  110  
1  1100  3266   55  51   0   4   0    0   60  115  
2   275   330    0   0   0   0   0    0   26   26  
3   445   584    5   5   0   0   0    0   21   26  
4   575   788    0   0   0   0   0    0   27   27  

[5 rows x 23 columns]

1월부터 12월까지 동일한 형식을 가지고 있습니다.

12개 월별 sheet를 한번에 모아줍니다.

In [7]:
seoulland_admission_2018=pd.DataFrame([])
for i in range(0,12):
    seoulland_admission_2018=pd.concat([seoulland_admission_2018
                                        ,pd.read_excel('Data/서울대공원 입장객 정보_2018년.xls'
                                        ,header=[1,3],sheet_name=i)],axis=0)    
    print(seoulland_admission_2018.shape)
                                        

(32, 23)
(61, 23)
(93, 23)
(124, 23)
(156, 23)
(187, 23)
(219, 23)
(251, 23)
(282, 23)
(314, 23)
(345, 23)
(377, 23)


In [8]:
seoulland_admission_2018

구분              월별입장객                     동,식물원         \
                     날짜   요일     날씨   유료합계   무료합계     총계 외국인계   유료소계     어른   
0   2018-01-01 00:00:00    월     맑음   1688    628   2316   40   1683   1375   
1   2018-01-02 00:00:00    화  구름 조금    403    585    988   22    401    342   
2   2018-01-03 00:00:00    수  구름 조금    192    502    694    4    191    153   
3   2018-01-04 00:00:00    목     흐림    165    506    671   14    163    132   
4   2018-01-05 00:00:00    금  구름 조금    189    569    758    8    188    145   
..                  ...  ...    ...    ...    ...    ...  ...    ...    ...   
27  2018-12-28 00:00:00    금     맑음     70    287    357   14     70     59   
28  2018-12-29 00:00:00    토     맑음    238    496    734   27    235    195   
29  2018-12-30 00:00:00    일     맑음    449    466    915   37    445    368   
30  2018-12-31 00:00:00    월     맑음    390    444    834   17    380    333   
31                   월계  NaN    NaN  21187  19058  40245  489  20963  15939   

          ...               테마가든                                  
     청소년  ...   무료소계     총계 유료소계   어른 청소년 어린이 외국인 단체입장 무료소계   총계  
0     99  ...    619   2302    5    5   0   0   0    0    9   14  
1     15  ...    578    979    2    2   0   0   0    0    7    9  
2     22  ...    502    693    1    0   0   1   0    0    0    1  
3      9  ...    488    651    2    2   0   0   0    0   18   20  
4     12  ...    567    755    1    1   0   0   0    0    2    3  
..   ...  ...    ...    ...  ...  ...  ..  ..  ..  ...  ...  ...  
27     7  ...    258    328    0    0   0   0   0    0   29   29  
28    13  ...    473    708    3    3   0   0   0    0   23   26  
29    21  ...    436    881    4    3   0   1   0    0   30   34  
30    22  ...    427    807   10   10   0   0   2    0   17   27  
31  2586  ...  18312  39275  224  206   3  15   2    0  746  970  

[377 rows x 23 columns]

1년은 365일인데, 377행입니다. 월계가 12개 들어가서 그렇습니다. '월계'는 삭제합니다.

In [9]:
seoulland_admission_2018=seoulland_admission_2018.loc[seoulland_admission_2018[('구분','날짜')]!='월계',:]

seoulland_admission_2018

구분           월별입장객                 동,식물원            ...  \
                     날짜 요일     날씨  유료합계 무료합계    총계 외국인계  유료소계    어른 청소년  ...   
0   2018-01-01 00:00:00  월     맑음  1688  628  2316   40  1683  1375  99  ...   
1   2018-01-02 00:00:00  화  구름 조금   403  585   988   22   401   342  15  ...   
2   2018-01-03 00:00:00  수  구름 조금   192  502   694    4   191   153  22  ...   
3   2018-01-04 00:00:00  목     흐림   165  506   671   14   163   132   9  ...   
4   2018-01-05 00:00:00  금  구름 조금   189  569   758    8   188   145  12  ...   
..                  ... ..    ...   ...  ...   ...  ...   ...   ...  ..  ...   
26  2018-12-27 00:00:00  목     맑음  3498  445  3943   10  3498  2960  33  ...   
27  2018-12-28 00:00:00  금     맑음    70  287   357   14    70    59   7  ...   
28  2018-12-29 00:00:00  토     맑음   238  496   734   27   235   195  13  ...   
29  2018-12-30 00:00:00  일     맑음   449  466   915   37   445   368  21  ...   
30  2018-12-31 00:00:00  월     맑음   390  444   834   17   380   333  22  ...   

              테마가든                                
   무료소계    총계 유료소계  어른 청소년 어린이 외국인 단체입장 무료소계  총계  
0   619  2302    5   5   0   0   0    0    9  14  
1   578   979    2   2   0   0   0    0    7   9  
2   502   693    1   0   0   1   0    0    0   1  
3   488   651    2   2   0   0   0    0   18  20  
4   567   755    1   1   0   0   0    0    2   3  
..  ...   ...  ...  ..  ..  ..  ..  ...  ...  ..  
26  418  3916    0   0   0   0   0    0   27  27  
27  258   328    0   0   0   0   0    0   29  29  
28  473   708    3   3   0   0   0    0   23  26  
29  436   881    4   3   0   1   0    0   30  34  
30  427   807   10  10   0   0   2    0   17  27  

[365 rows x 23 columns]

쓰지 않을 데이터는 미리 정리하겠습니다.

우선 월별입장액은 동,식물원과 테마가든의 합계인데, 어른, 청소년, 어린이 등의 구분이 없습니다.

테마가든은 동,식물원에 비해 수치가 미미해서 크게 영향이 없으므로 동,식물원만 보겠습니다.

In [10]:
seoulland_admission_2018=seoulland_admission_2018.iloc[:,[0,1,2,7,8,9,10,11,12,13,14]]
seoulland_admission_2018.head()

구분           동,식물원                                   
                    날짜 요일     날씨  유료소계    어른 청소년  어린이 외국인 단체입장 무료소계    총계
0  2018-01-01 00:00:00  월     맑음  1683  1375  99  209  40    0  619  2302
1  2018-01-02 00:00:00  화  구름 조금   401   342  15   44  22  119  578   979
2  2018-01-03 00:00:00  수  구름 조금   191   153  22   16   4    0  502   693
3  2018-01-04 00:00:00  목     흐림   163   132   9   22  14    1  488   651
4  2018-01-05 00:00:00  금  구름 조금   188   145  12   31   8    0  567   755

컬럼명이 1레벨에 '구분', '동,식물원' 이고, 2레벨에서 '날짜','요일','날씨' 등으로 이루어져있습니다. 이런 것을 멀티인덱스라고 하는데, 멀티인덱스는 다루기 불편하니, 단일 인덱스로 수정하겠습니다.

In [11]:
seoulland_admission_2018.columns=['날짜','요일','날씨','유료소계','어른','청소년','어린이','외국인','단체입장','무료소계','총계']

In [12]:
seoulland_admission_2018.head()

날짜 요일     날씨  유료소계    어른  청소년  어린이  외국인  단체입장  무료소계    총계
0  2018-01-01 00:00:00  월     맑음  1683  1375   99  209   40     0   619  2302
1  2018-01-02 00:00:00  화  구름 조금   401   342   15   44   22   119   578   979
2  2018-01-03 00:00:00  수  구름 조금   191   153   22   16    4     0   502   693
3  2018-01-04 00:00:00  목     흐림   163   132    9   22   14     1   488   651
4  2018-01-05 00:00:00  금  구름 조금   188   145   12   31    8     0   567   755

## 2019년 데이터 로딩

In [13]:
seoulland_admission_2019_1=pd.DataFrame([])
for i in range(0,12):
    seoulland_admission_2019_1=pd.concat([seoulland_admission_2019_1
                             ,pd.read_excel('Data/서울대공원 입장객 정보_2019년.xls',header=[1,3],sheet_name=i)]
                             ,axis=0)    
    print(seoulland_admission_2019_1.shape)


(32, 23)
(61, 23)
(93, 23)
(124, 23)
(155, 31)
(185, 35)
(216, 35)
(247, 35)
(276, 35)
(307, 35)
(337, 35)
(368, 35)


In [14]:
seoulland_admission_2019_1

구분                         동,식물원                                    \
       날씨                   날짜 요일  단체입장 단체입장.1  단체입장.2   무료소계    어른   어린이   
0   구름 조금  2019-01-01 00:00:00  화     0    NaN     NaN  581.0   770    94   
1      맑음  2019-01-02 00:00:00  수     0    NaN     NaN  446.0   147    14   
2      맑음  2019-01-03 00:00:00  목     0    NaN     NaN  579.0   153    21   
3   구름 많음  2019-01-04 00:00:00  금     0    NaN     NaN  484.0   139    23   
4     NaN  2019-01-05 00:00:00  토     0    NaN     NaN  842.0   615    87   
..    ...                  ... ..   ...    ...     ...    ...   ...   ...   
26     맑음  2019-12-27 00:00:00  금     0  455.0  1271.0    NaN   274    23   
27     흐림  2019-12-28 00:00:00  토    40  997.0  2046.0    NaN   862   140   
28    NaN  2019-12-29 00:00:00  일  5066  807.0  6866.0    NaN  4585  1301   
29    NaN  2019-12-30 00:00:00  월     0  429.0   840.0    NaN   347    42   
30    NaN  2019-12-31 00:00:00  화     0  344.0   529.0    NaN   164    14   

         ... 테마가든                                                 
    외국인  ... 단체입장 단체입장.1 단체입장.2  무료소계  어른 어린이 외국인 유료소계 청소년    총계  
0    54  ...    0    NaN    NaN  24.0   6   1   0    7   0  31.0  
1    11  ...    0    NaN    NaN  13.0   0   0   0    0   0  13.0  
2    16  ...    0    NaN    NaN  16.0   0   0   0    0   0  16.0  
3     3  ...    0    NaN    NaN  25.0   5   0   0    5   0  30.0  
4    16  ...    0    NaN    NaN  19.0   6   0   0    6   0  25.0  
..  ...  ...  ...    ...    ...   ...  ..  ..  ..  ...  ..   ...  
26  400  ...    0   28.0   31.0   NaN   2   1   4    3   0   NaN  
27  400  ...    0   39.0   64.0   NaN  24   1   4   25   0   NaN  
28  400  ...    0   39.0   47.0   NaN   5   3   4    8   0   NaN  
29  400  ...    0   21.0   23.0   NaN   2   0   4    2   0   NaN  
30  400  ...    0   15.0   19.0   NaN   4   0   4    4   0   NaN  

[368 rows x 35 columns]

다르게 생겼습니다..ㅠㅠ 이때는 원본 엑셀파일 열어보는 것이 가장 빠릅니다. 찾아보니, 5월부터 데이터가 이상합니다. 4월까지 하나 만들고, 그 이후를 하나로 만들겠습니다.

In [15]:
seoulland_admission_2019_1=pd.DataFrame([])
for i in range(0,4):
    seoulland_admission_2019_1=pd.concat([seoulland_admission_2019_1
                             ,pd.read_excel('Data/서울대공원 입장객 정보_2019년.xls',header=[1,3],sheet_name=i)]
                             ,axis=0)    
    print(seoulland_admission_2019_1.shape)

(32, 23)
(61, 23)
(93, 23)
(124, 23)


5월부터는 무엇이 문제인지 봅시다.

In [16]:
seoulland_admission_2019_2=pd.DataFrame([])
for i in range(4,12):
    seoulland_admission_2019_2=pd.concat([seoulland_admission_2019_2
                                          ,pd.read_excel('Data/서울대공원 입장객 정보_2019년.xls'
                                                        ,header=[1,2,3],sheet_name=i)],axis=0)    
    print(seoulland_admission_2019_2.shape)
seoulland_admission_2019_2

(31, 23)
(61, 27)
(92, 27)
(123, 27)
(152, 27)
(183, 27)
(213, 27)
(244, 27)


구분                                        \
   Unnamed: 0_level_1 Unnamed: 1_level_1 Unnamed: 2_level_1   
                   날짜                 요일                 날씨   
0          2019-05-01                  수                NaN   
1          2019-05-02                  목                NaN   
2          2019-05-03                  금                 맑음   
3          2019-05-04                  토              구름 조금   
4          2019-05-05                  일              구름 조금   
..                ...                ...                ...   
26         2019-12-27                  금                 맑음   
27         2019-12-28                  토                 흐림   
28         2019-12-29                  일                NaN   
29         2019-12-30                  월                NaN   
30         2019-12-31                  화                NaN   

                 동,식물원                                          \
   Unnamed: 10_level_1 Unnamed: 11_level_1 Unnamed: 12_level_1   
                   어린이                 외국인                단체입장   
0                 2027                1839                1476   
1                  449                1839                1254   
2                  654                1839                2830   
3                 1758                1839                 104   
4                    0                1839                4792   
..                 ...                 ...                 ...   
26                  23                 400                   0   
27                 140                 400                  40   
28                1301                 400                5066   
29                  42                 400                   0   
30                  14                 400                   0   

                                                             \
   Unnamed: 7_level_1 Unnamed: 8_level_1 Unnamed: 9_level_1   
                 유료소계                 어른                청소년   
0               18510              15183               1300   
1                4081               2485               1147   
2                6031               3202               2175   
3               12140              10010                372   
4               20451              19854                597   
..                ...                ...                ...   
26                816                274                519   
27               1049                862                 47   
28               6059               4585                173   
29                411                347                 22   
30                185                164                  7   

                        ...           월별입장객 현황                     \
                  무료소계  ...               유료합계                 총계   
   Unnamed: 13_level_2  ... Unnamed: 3_level_2 Unnamed: 5_level_2   
0                 6096  ...            19887.0            26327.0   
1                 1854  ...             4446.0             6977.0   
2                 2398  ...             6433.0             9589.0   
3                 4148  ...            12766.0            17366.0   
4                20856  ...            21667.0            43255.0   
..                 ...  ...                ...                ...   
26                 455  ...              819.0             1302.0   
27                 997  ...             1074.0             2110.0   
28                 807  ...             6067.0             6913.0   
29                 429  ...              413.0              863.0   
30                 344  ...              189.0              548.0   

                  테마가든                                          \
   Unnamed: 15_level_1 Unnamed: 16_level_1 Unnamed: 17_level_1   
                  유료소계                  어른                 청소년   
0                 1377                1256                  12   
1                  365                 252                   3   
2                  402         

열수가 4개 더 많습니다. 

엑셀을 잘 살펴보니, "월별 입장객 현황"과 "월별입장객 현황"이 차이가 있습니다. 6월만 "월별 입장객 현황" 입니다.

일단 6월을 빼고 합쳐줍니다.

In [17]:
seoulland_admission_2019_2=pd.DataFrame([])
for i in range(4,12):
    if i!=5:
        seoulland_admission_2019_2=pd.concat([seoulland_admission_2019_2
                                              ,pd.read_excel('Data/서울대공원 입장객 정보_2019년.xls'
                                                             ,header=[1,2,3],sheet_name=i)],axis=0)    
    print(seoulland_admission_2019_2.shape)
seoulland_admission_2019_2

(31, 23)
(31, 23)
(62, 23)
(93, 23)
(122, 23)
(153, 23)
(183, 23)
(214, 23)


구분                                        \
   Unnamed: 0_level_1 Unnamed: 1_level_1 Unnamed: 2_level_1   
                   날짜                 요일                 날씨   
0          2019-05-01                  수                NaN   
1          2019-05-02                  목                NaN   
2          2019-05-03                  금                 맑음   
3          2019-05-04                  토              구름 조금   
4          2019-05-05                  일              구름 조금   
..                ...                ...                ...   
26         2019-12-27                  금                 맑음   
27         2019-12-28                  토                 흐림   
28         2019-12-29                  일                NaN   
29         2019-12-30                  월                NaN   
30         2019-12-31                  화                NaN   

             월별입장객 현황                                        \
                 유료합계               무료합계                 총계   
   Unnamed: 3_level_2 Unnamed: 4_level_2 Unnamed: 5_level_2   
0               19887               6440              26327   
1                4446               2531               6977   
2                6433               3156               9589   
3               12766               4600              17366   
4               21667              21588              43255   
..                ...                ...                ...   
26                819                483               1302   
27               1074               1036               2110   
28               6067                846               6913   
29                413                450                863   
30                189                359                548   

                                   동,식물원                     \
                 외국인계 Unnamed: 7_level_1 Unnamed: 8_level_1   
   Unnamed: 6_level_2               유료소계                 어른   
0                2307              18510              15183   
1                2307               4081               2485   
2                2307               6031               3202   
3                2307              12140              10010   
4                2307              20451              19854   
..                ...                ...                ...   
26                404                816                274   
27                404               1049                862   
28                404               6059               4585   
29                404                411                347   
30                404                185                164   

                       ...                                          \
   Unnamed: 9_level_1  ...                무료소계                  총계   
                  청소년  ... Unnamed: 13_level_2 Unnamed: 14_level_2   
0                1300  ...                6096               24606   
1                1147  ...                1854                5935   
2                2175  ...                2398                8429   
3                 372  ...                4148               16288   
4                 597  ...               20856               41307   
..                ...  ...                 ...                 ...   
26                519  ...                 455                1271   
27                 47  ...                 997                2046   
28                173  ...                 807                6866   
29                 22  ...                 429                 840   
30                  7  ...                 344                 529   

                  테마가든                                          \
   Unnamed: 15_level_1 Unnamed: 16_level_1 Unnamed: 17_level_1   
                  유료소계                  어른                 청소년   
0                 1377                1256                  12   
1                  365                 252                   3   
2                  402                 238                 

In [18]:
seoulland_admission_2019_2

구분                                        \
   Unnamed: 0_level_1 Unnamed: 1_level_1 Unnamed: 2_level_1   
                   날짜                 요일                 날씨   
0          2019-05-01                  수                NaN   
1          2019-05-02                  목                NaN   
2          2019-05-03                  금                 맑음   
3          2019-05-04                  토              구름 조금   
4          2019-05-05                  일              구름 조금   
..                ...                ...                ...   
26         2019-12-27                  금                 맑음   
27         2019-12-28                  토                 흐림   
28         2019-12-29                  일                NaN   
29         2019-12-30                  월                NaN   
30         2019-12-31                  화                NaN   

             월별입장객 현황                                        \
                 유료합계               무료합계                 총계   
   Unnamed: 3_level_2 Unnamed: 4_level_2 Unnamed: 5_level_2   
0               19887               6440              26327   
1                4446               2531               6977   
2                6433               3156               9589   
3               12766               4600              17366   
4               21667              21588              43255   
..                ...                ...                ...   
26                819                483               1302   
27               1074               1036               2110   
28               6067                846               6913   
29                413                450                863   
30                189                359                548   

                                   동,식물원                     \
                 외국인계 Unnamed: 7_level_1 Unnamed: 8_level_1   
   Unnamed: 6_level_2               유료소계                 어른   
0                2307              18510              15183   
1                2307               4081               2485   
2                2307               6031               3202   
3                2307              12140              10010   
4                2307              20451              19854   
..                ...                ...                ...   
26                404                816                274   
27                404               1049                862   
28                404               6059               4585   
29                404                411                347   
30                404                185                164   

                       ...                                          \
   Unnamed: 9_level_1  ...                무료소계                  총계   
                  청소년  ... Unnamed: 13_level_2 Unnamed: 14_level_2   
0                1300  ...                6096               24606   
1                1147  ...                1854                5935   
2                2175  ...                2398                8429   
3                 372  ...                4148               16288   
4                 597  ...               20856               41307   
..                ...  ...                 ...                 ...   
26                519  ...                 455                1271   
27                 47  ...                 997                2046   
28                173  ...                 807                6866   
29                 22  ...                 429                 840   
30                  7  ...                 344                 529   

                  테마가든                                          \
   Unnamed: 15_level_1 Unnamed: 16_level_1 Unnamed: 17_level_1   
                  유료소계                  어른                 청소년   
0                 1377                1256                  12   
1                  365                 252                   3   
2                  402                 238                 

6월만 따로 저장합니다.

In [19]:
seoulland_admission_2019_3=pd.read_excel('Data/서울대공원 입장객 정보_2019년.xls',header=[1,2,3],sheet_name=5)

print(seoulland_admission_2019_3.shape)

seoulland_admission_2019_3


(30, 23)


구분                                        \
   Unnamed: 0_level_1 Unnamed: 1_level_1 Unnamed: 2_level_1   
                   날짜                 요일                 날씨   
0          2019-06-01                  토              구름 많음   
1          2019-06-02                  일                 맑음   
2          2019-06-03                  월                 맑음   
3          2019-06-04                  화                NaN   
4          2019-06-05                  수                NaN   
5          2019-06-06                  목                NaN   
6          2019-06-07                  금                 흐림   
7          2019-06-08                  토                 맑음   
8          2019-06-09                  일                  비   
9          2019-06-10                  월                 흐림   
10         2019-06-11                  화              구름 많음   
11         2019-06-12                  수                 맑음   
12         2019-06-13                  목                 맑음   
13         2019-06-14                  금              구름 많음   
14         2019-06-15                  토                  비   
15         2019-06-16                  일                 맑음   
16         2019-06-17                  월                 맑음   
17         2019-06-18                  화                  비   
18         2019-06-19                  수                NaN   
19         2019-06-20                  목                NaN   
20         2019-06-21                  금                NaN   
21         2019-06-22                  토                NaN   
22         2019-06-23                  일                 맑음   
23         2019-06-24                  월                 맑음   
24         2019-06-25                  화                 맑음   
25         2019-06-26                  수              구름 많음   
26         2019-06-27                  목              구름 많음   
27         2019-06-28                  금                소나기   
28         2019-06-29                  토                 흐림   
29         2019-06-30                  일                NaN   

            월별 입장객 현황                                        \
                 유료합계               무료합계                 총계   
   Unnamed: 3_level_2 Unnamed: 4_level_2 Unnamed: 5_level_2   
0               19216               7148              26364   
1               17606               7843              25449   
2                3090               2966               6056   
3                3196               3332               6528   
4                3815               3160               6975   
5               14809               7153              21962   
6                1597               1015               2612   
7               18388               8787              27175   
8                9791               5734              15525   
9                1241               1832               3073   
10               1674               3059               4733   
11               2592               2200               4792   
12               1280               2081               3361   
13               1481               1960               3441   
14               8612               4312              12924   
15              10391               4289              14680   
16               1719               1428               3147   
17                843               1242               2085   
18                917               1295               2212   
19               1227               2044               3271   
20               1009               1448               2457   
21               8684               3917              12601   
22               8066               3223              11289   
23               1807               1114               2921   
24                810               1228               2038   
25                677               1277               1954   
26                650                937               1587   
27              16017               1109         

컬럼의 수가 같아졌으니, 컬럼명을 1~4월과 동일하게 맞춰줍니다.

In [20]:
seoulland_admission_2019_2.columns=seoulland_admission_2019_1.columns

seoulland_admission_2019_3.columns=seoulland_admission_2019_1.columns

이제 3개 데이터프레임을 합쳐줍니다

In [21]:
seoulland_admission_2019=pd.concat([seoulland_admission_2019_1,seoulland_admission_2019_2,seoulland_admission_2019_3])
seoulland_admission_2019

구분            월별입장객                     동,식물원         \
                     날짜 요일     날씨   유료합계  무료합계     총계  외국인계   유료소계     어른   
0   2019-01-01 00:00:00  화  구름 조금    919   605   1524    54    912    770   
1   2019-01-02 00:00:00  수     맑음    175   459    634    11    175    147   
2   2019-01-03 00:00:00  목     맑음    182   595    777    16    182    153   
3   2019-01-04 00:00:00  금  구름 많음    181   509    690     3    176    139   
4   2019-01-05 00:00:00  토    NaN    740   861   1601    16    734    615   
..                  ... ..    ...    ...   ...    ...   ...    ...    ...   
25  2019-06-26 00:00:00  수  구름 많음    677  1277   1954  2167    608    564   
26  2019-06-27 00:00:00  목  구름 많음    650   937   1587  2167    610    549   
27  2019-06-28 00:00:00  금    소나기  16017  1109  17126  2167  15948  10570   
28  2019-06-29 00:00:00  토     흐림   3108  1635   4743  2167   2834   2454   
29  2019-06-30 00:00:00  일    NaN   6680  2742   9422  2167   6062   5320   

         ...              테마가든                                   
    청소년  ...  무료소계     총계 유료소계   어른 청소년 어린이  외국인 단체입장 무료소계   총계  
0    48  ...   581   1493    7    6   0   1    0    0   24   31  
1    14  ...   446    621    0    0   0   0    0    0   13   13  
2     8  ...   579    761    0    0   0   0    0    0   16   16  
3    14  ...   484    660    5    5   0   0    0    0   25   30  
4    32  ...   842   1576    6    6   0   0    0    0   19   25  
..  ...  ...   ...    ...  ...  ...  ..  ..  ...  ...  ...  ...  
25   12  ...  1060   1668   69   66   1   2  726  119  217  286  
26   30  ...   759   1369   40   40   0   0  726    0  178  218  
27  707  ...   887  16835   69   59   3   7  726  107  222  291  
28   41  ...  1486   4320  274  245   3  26  726    0  149  423  
29   61  ...  2381   8443  618  559   3  56  726    0  361  979  

[368 rows x 23 columns]

In [22]:
#사용할 컬럼만 남기기
seoulland_admission_2019=seoulland_admission_2019.iloc[:,[0,1,2,7,8,9,10,11,12,13,14]]
#컬럼명 수정
seoulland_admission_2019.columns=seoulland_admission_2018.columns
#'월계'를 제거합니다.
seoulland_admission_2019=seoulland_admission_2019.loc[seoulland_admission_2019['날짜']!='월계',:]

seoulland_admission_2019

날짜 요일     날씨   유료소계     어른  청소년   어린이   외국인   단체입장  무료소계  \
0   2019-01-01 00:00:00  화  구름 조금    912    770   48    94    54      0   581   
1   2019-01-02 00:00:00  수     맑음    175    147   14    14    11      0   446   
2   2019-01-03 00:00:00  목     맑음    182    153    8    21    16      0   579   
3   2019-01-04 00:00:00  금  구름 많음    176    139   14    23     3      0   484   
4   2019-01-05 00:00:00  토    NaN    734    615   32    87    16      0   842   
..                  ... ..    ...    ...    ...  ...   ...   ...    ...   ...   
25  2019-06-26 00:00:00  수  구름 많음    608    564   12    32  1441    147  1060   
26  2019-06-27 00:00:00  목  구름 많음    610    549   30    31  1441      0   759   
27  2019-06-28 00:00:00  금    소나기  15948  10570  707  4671  1441  15345   887   
28  2019-06-29 00:00:00  토     흐림   2834   2454   41   339  1441    117  1486   
29  2019-06-30 00:00:00  일    NaN   6062   5320   61   681  1441     36  2381   

       총계  
0    1493  
1     621  
2     761  
3     660  
4    1576  
..    ...  
25   1668  
26   1369  
27  16835  
28   4320  
29   8443  

[364 rows x 11 columns]

365 rows여야 하는데, 하나가 빕니다.

행은 찾아보니, 19년 9월 7일이 비어있습니다(눈으로 찾음). 일단 행을 추가하겠습니다.

In [23]:
seoulland_admission_2019.iloc[218:220]

날짜 요일     날씨  유료소계    어른  청소년  어린이   외국인  단체입장  무료소계    총계
5  2019-09-06 00:00:00  금  구름 많음   485   271  171   43  2163   140   446   931
6  2019-09-08 00:00:00  일    NaN  4233  3722   54  457  2163     0  1453  5686

In [24]:
new_row=pd.DataFrame([['2019-09-07 00:00:00','토',np.nan,np.nan,np.nan,np.nan,np.nan,np.nan,np.nan,np.nan,np.nan]],columns=seoulland_admission_2019.columns)

In [25]:
new_row

날짜 요일  날씨  유료소계  어른  청소년  어린이  외국인  단체입장  무료소계  총계
0  2019-09-07 00:00:00  토 NaN   NaN NaN  NaN  NaN  NaN   NaN   NaN NaN

In [26]:
seoulland_admission_2019=pd.concat([seoulland_admission_2019,new_row])

In [27]:
seoulland_admission_2019

날짜 요일     날씨     유료소계       어른    청소년     어린이     외국인  \
0   2019-01-01 00:00:00  화  구름 조금    912.0    770.0   48.0    94.0    54.0   
1   2019-01-02 00:00:00  수     맑음    175.0    147.0   14.0    14.0    11.0   
2   2019-01-03 00:00:00  목     맑음    182.0    153.0    8.0    21.0    16.0   
3   2019-01-04 00:00:00  금  구름 많음    176.0    139.0   14.0    23.0     3.0   
4   2019-01-05 00:00:00  토    NaN    734.0    615.0   32.0    87.0    16.0   
..                  ... ..    ...      ...      ...    ...     ...     ...   
26  2019-06-27 00:00:00  목  구름 많음    610.0    549.0   30.0    31.0  1441.0   
27  2019-06-28 00:00:00  금    소나기  15948.0  10570.0  707.0  4671.0  1441.0   
28  2019-06-29 00:00:00  토     흐림   2834.0   2454.0   41.0   339.0  1441.0   
29  2019-06-30 00:00:00  일    NaN   6062.0   5320.0   61.0   681.0  1441.0   
0   2019-09-07 00:00:00  토    NaN      NaN      NaN    NaN     NaN     NaN   

       단체입장    무료소계       총계  
0       0.0   581.0   1493.0  
1       0.0   446.0    621.0  
2       0.0   579.0    761.0  
3       0.0   484.0    660.0  
4       0.0   842.0   1576.0  
..      ...     ...      ...  
26      0.0   759.0   1369.0  
27  15345.0   887.0  16835.0  
28    117.0  1486.0   4320.0  
29     36.0  2381.0   8443.0  
0       NaN     NaN      NaN  

[365 rows x 11 columns]

나중에 날짜순으로 정렬해 줄 예정이니, 순서는 일단 내버려둡니다.

## 2020년 데이터 로딩

In [28]:
seoulland_admission_2020=pd.DataFrame([])
for i in range(0,12):
    seoulland_admission_2020=pd.concat([seoulland_admission_2020,pd.read_excel('Data/서울대공원 입장객 정보_2020년.xlsx',header=[1,2,3],engine='openpyxl',sheet_name=i)],axis=0)
    print(seoulland_admission_2020.shape)

(31, 23)
(60, 23)
(91, 23)
(121, 23)


ParserError: Passed header=[1,2,3] are too many rows for this multi_index of columns

i가 3까지 가고 중단되었습니다.

일단 4월까지 합칩니다.

In [29]:
seoulland_admission_2020_1=pd.DataFrame([])
for i in range(0,4):
    seoulland_admission_2020_1=pd.concat([seoulland_admission_2020_1,pd.read_excel('Data/서울대공원 입장객 정보_2020년.xlsx',header=[1,2,3],engine='openpyxl',sheet_name=i)],axis=0)
    print(seoulland_admission_2020_1.shape)

(31, 23)
(60, 23)
(91, 23)
(121, 23)


In [30]:
seoulland_admission_2020_1

구분                                        \
   Unnamed: 0_level_1 Unnamed: 1_level_1 Unnamed: 2_level_1   
                   날짜                 요일                 날씨   
0          2020-01-01                  수                 흐림   
1          2020-01-02                  목                 맑음   
2          2020-01-03                  금                 맑음   
3          2020-01-04                  토                 맑음   
4          2020-01-05                  일                 맑음   
..                ...                ...                ...   
25         2020-04-26                  일              구름 많음   
26         2020-04-27                  월              구름 많음   
27         2020-04-28                  화                 맑음   
28         2020-04-29                  수              구름 많음   
29         2020-04-30                  목                 맑음   

             월별입장객 현황                                        \
                 유료합계               무료합계                 총계   
   Unnamed: 3_level_2 Unnamed: 4_level_2 Unnamed: 5_level_2   
0                 774                465               1239   
1                 316                550                866   
2                 321                491                812   
3                1128                931               2059   
4                1520               1201               2721   
..                ...                ...                ...   
25               9962               2056              12018   
26               1405                863               2268   
27               1862               1118               2980   
28               3271                985               4256   
29              13003               3595              16598   

                                   동,식물원                     \
                 외국인계 Unnamed: 7_level_1 Unnamed: 8_level_1   
   Unnamed: 6_level_2               유료소계                 어른   
0                 772                767                650   
1                 772                315                279   
2                 772                318                264   
3                 772               1117                935   
4                 772               1500               1281   
..                ...                ...                ...   
25                218               9406               8128   
26                218               1359               1227   
27                218               1766               1564   
28                218               3186               2642   
29                218              12374              10501   

                       ...                                          \
   Unnamed: 9_level_1  ...                무료소계                  총계   
                  청소년  ... Unnamed: 13_level_2 Unnamed: 14_level_2   
0                  30  ...                 452                1219   
1                  11  ...                 528                 843   
2                  22  ...                 439                 757   
3                  42  ...                 915                2032   
4                  30  ...                1167                2667   
..                ...  ...                 ...                 ...   
25                185  ...                1774               11180   
26                 23  ...                 745                2104   
27                 19  ...                 943                2709   
28                 71  ...                 814                4000   
29                276  ...                3319               15693   

                  테마가든                                          \
   Unnamed: 15_level_1 Unnamed: 16_level_1 Unnamed: 17_level_1   
                  유료소계                  어른                 청소년   
0                    7                   6                   0   
1                    1                   0                   0   
2                    3                   2                 

이제 5월부터 나머지를 합칩니다.

In [31]:
seoulland_admission_2020_2=pd.DataFrame([])
for i in range(4,12):
    seoulland_admission_2020_2=pd.concat([seoulland_admission_2020_2,pd.read_excel('Data/서울대공원 입장객 정보_2020년.xlsx',header=[4],engine='openpyxl',sheet_name=i)],axis=0)
    print(seoulland_admission_2020_2.shape)

(32, 25)
(63, 25)
(95, 25)
(127, 25)
(158, 25)
(190, 25)
(220, 48)
(251, 68)


11월에 컬럼수가 늘어납니다. 5~10월까지 묶습니다.

In [32]:
seoulland_admission_2020_2=pd.DataFrame([])
for i in range(4,10):
    seoulland_admission_2020_2=pd.concat([seoulland_admission_2020_2,pd.read_excel('Data/서울대공원 입장객 정보_2020년.xlsx',header=[4],engine='openpyxl',sheet_name=i)],axis=0)
    print(seoulland_admission_2020_2.shape)

(32, 25)
(63, 25)
(95, 25)
(127, 25)
(158, 25)
(190, 25)


In [33]:
seoulland_admission_2020_2.head()

날짜    요일     기타    유료일계   무료일계     일합계  외국인계     유료계  \
0                   월계  None   None  124580  52711  177291   566  119678   
1  2020-05-01 00:00:00     금     맑음   12170   3563   15733    54   11854   
2  2020-05-02 00:00:00     토     흐림    9814   3783   13597    65    9547   
3  2020-05-03 00:00:00     일     흐림    8879   1680   10559    87    8480   
4  2020-05-04 00:00:00     월  구름 많음    7927   2073   10000    19    7768   

       어른   청소년  ...  유료계.1  어른.1  청소년.1  어린이.1  외국인.1  단체입장.1  Unnamed: 21  \
0  104249  2379  ...   4902  4442     60    400     21       0         3661   
1   10532   163  ...    316   280     11     25      5       0          252   
2    8141   202  ...    267   228      3     36      0       0          286   
3    7195   200  ...    399   350     11     38      3       0          193   
4    6799   114  ...    159   149      0     10      2       0          137   

   Unnamed: 22  Unnamed: 23  Unnamed: 24  
0         8563          NaN          NaN  
1          568          NaN          NaN  
2          553          NaN          NaN  
3          592          NaN          NaN  
4          296          NaN          NaN  

[5 rows x 25 columns]

In [34]:
seoulland_admission_2020_2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 190 entries, 0 to 31
Data columns (total 25 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   날짜           190 non-null    object 
 1   요일           184 non-null    object 
 2   기타           123 non-null    object 
 3   유료일계         190 non-null    int64  
 4   무료일계         190 non-null    int64  
 5   일합계          190 non-null    int64  
 6   외국인계         190 non-null    int64  
 7   유료계          190 non-null    int64  
 8   어른           190 non-null    int64  
 9   청소년          190 non-null    int64  
 10  어린이          190 non-null    int64  
 11  외국인          190 non-null    int64  
 12  단체입장         190 non-null    int64  
 13  Unnamed: 13  190 non-null    int64  
 14  Unnamed: 14  190 non-null    int64  
 15  유료계.1        190 non-null    int64  
 16  어른.1         190 non-null    int64  
 17  청소년.1        190 non-null    int64  
 18  어린이.1        190 non-null    int64  
 19  외국인.1    

이제 11~12월을 묶습니다.

In [35]:
seoulland_admission_2020_3=pd.DataFrame([])
for i in range(10,12):
    seoulland_admission_2020_3=pd.concat([seoulland_admission_2020_3,pd.read_excel('Data/서울대공원 입장객 정보_2020년.xlsx',header=[4],engine='openpyxl',sheet_name=i)],axis=0)
    print(seoulland_admission_2020_3.shape)

(30, 25)
(61, 45)


그와중에 또 다르네요.

In [36]:
seoulland_admission_2020_3=pd.read_excel('Data/서울대공원 입장객 정보_2020년.xlsx',header=[3],engine='openpyxl',sheet_name=10)
print(seoulland_admission_2020_3.shape)
seoulland_admission_2020_4=pd.read_excel('Data/서울대공원 입장객 정보_2020년.xlsx',header=[3],engine='openpyxl',sheet_name=11)
print(seoulland_admission_2020_4.shape)

(31, 25)
(32, 25)


In [37]:
seoulland_admission_2020_1=seoulland_admission_2020_1.iloc[:,[0,1,2,7,8,9,10,11,12,13,14]]
seoulland_admission_2020_2=seoulland_admission_2020_2.iloc[:,[0,1,2,7,8,9,10,11,12,13,14]]
seoulland_admission_2020_3=seoulland_admission_2020_3.iloc[:,[0,1,2,7,8,9,10,11,12,13,14]]
seoulland_admission_2020_4=seoulland_admission_2020_4.iloc[:,[0,1,2,7,8,9,10,11,12,13,14]]

seoulland_admission_2020_1.columns=seoulland_admission_2018.columns
seoulland_admission_2020_2.columns=seoulland_admission_2018.columns
seoulland_admission_2020_3.columns=seoulland_admission_2018.columns
seoulland_admission_2020_4.columns=seoulland_admission_2018.columns

In [38]:
seoulland_admission_2020=pd.concat([seoulland_admission_2020_1,seoulland_admission_2020_2,seoulland_admission_2020_3,seoulland_admission_2020_4])
seoulland_admission_2020

날짜 요일   날씨  유료소계    어른  청소년  어린이  외국인  단체입장  무료소계    총계
0   2020-01-01 00:00:00  수   흐림   767   650   30   87  764     0   452  1219
1   2020-01-02 00:00:00  목   맑음   315   279   11   25  764     0   528   843
2   2020-01-03 00:00:00  금   맑음   318   264   22   32  764     0   439   757
3   2020-01-04 00:00:00  토   맑음  1117   935   42  140  764     0   915  2032
4   2020-01-05 00:00:00  일   맑음  1500  1281   30  189  764     0  1167  2667
..                  ... ..  ...   ...   ...  ...  ...  ...   ...   ...   ...
27  2020-12-27 00:00:00  일  NaN  1187  1019   32  136    9     0   451  1638
28  2020-12-28 00:00:00  월  NaN   497   450   12   35    2     0   223   720
29  2020-12-29 00:00:00  화  NaN   155   137    1   17    0     0    64   219
30  2020-12-30 00:00:00  수  NaN    39    35    0    4    0     0    22    61
31  2020-12-31 00:00:00  목  NaN    47    42    3    2    0     0    54   101

[374 rows x 11 columns]

In [39]:
seoulland_admission_2020=seoulland_admission_2020.loc[~seoulland_admission_2020['날짜'].isin(['월 총계','월계']),:]

seoulland_admission_2020

날짜 요일   날씨  유료소계    어른  청소년  어린이  외국인  단체입장  무료소계    총계
0   2020-01-01 00:00:00  수   흐림   767   650   30   87  764     0   452  1219
1   2020-01-02 00:00:00  목   맑음   315   279   11   25  764     0   528   843
2   2020-01-03 00:00:00  금   맑음   318   264   22   32  764     0   439   757
3   2020-01-04 00:00:00  토   맑음  1117   935   42  140  764     0   915  2032
4   2020-01-05 00:00:00  일   맑음  1500  1281   30  189  764     0  1167  2667
..                  ... ..  ...   ...   ...  ...  ...  ...   ...   ...   ...
27  2020-12-27 00:00:00  일  NaN  1187  1019   32  136    9     0   451  1638
28  2020-12-28 00:00:00  월  NaN   497   450   12   35    2     0   223   720
29  2020-12-29 00:00:00  화  NaN   155   137    1   17    0     0    64   219
30  2020-12-30 00:00:00  수  NaN    39    35    0    4    0     0    22    61
31  2020-12-31 00:00:00  목  NaN    47    42    3    2    0     0    54   101

[366 rows x 11 columns]

2020년은 366이 맞습니다.

In [40]:
seoulland_admission_2020_2=seoulland_admission_2020_2.loc[~seoulland_admission_2020_2['날짜'].isin(['월 총계','월계']),:]

## 2021년 데이터 로딩

In [67]:
seoulland_admission_2021=pd.DataFrame([])
for i in range(0,12):
    seoulland_admission_2021=pd.concat([seoulland_admission_2021,pd.read_excel('Data/서울대공원 입장객 정보_2021년.xlsx',header=[4],engine='openpyxl',sheet_name=i)],axis=0)
    print(seoulland_admission_2021.shape)

(32, 25)
(61, 25)
(93, 25)
(124, 25)
(156, 25)
(187, 25)
(219, 25)
(251, 25)
(282, 25)
(314, 25)
(345, 25)
(378, 25)


다 맞습니다.

In [68]:
seoulland_admission_2021.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 378 entries, 0 to 32
Data columns (total 25 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   날짜           377 non-null    object 
 1   요일           365 non-null    object 
 2   기타           0 non-null      float64
 3   유료일계         377 non-null    float64
 4   무료일계         377 non-null    float64
 5   일합계          377 non-null    float64
 6   외국인계         377 non-null    float64
 7   유료계          377 non-null    float64
 8   어른           377 non-null    float64
 9   청소년          377 non-null    float64
 10  어린이          377 non-null    float64
 11  외국인          377 non-null    float64
 12  단체입장         376 non-null    float64
 13  Unnamed: 13  377 non-null    float64
 14  Unnamed: 14  377 non-null    float64
 15  유료계.1        377 non-null    float64
 16  어른.1         377 non-null    float64
 17  청소년.1        377 non-null    float64
 18  어린이.1        377 non-null    float64
 19  외국인.1    

In [69]:
#컬럼 선택
seoulland_admission_2021=seoulland_admission_2021.iloc[:,[0,1,2,7,8,9,10,11,12,13,14]]

#컬럼명 변경
seoulland_admission_2021.columns=seoulland_admission_2018.columns

#불필요한 행 삭제
seoulland_admission_2021=seoulland_admission_2021.loc[~seoulland_admission_2021['날짜'].isin(['월 총계','월계']),:]

In [70]:
seoulland_admission_2021

날짜   요일  날씨   유료소계     어른   청소년   어린이  외국인  단체입장   무료소계  \
1   2021-01-01 00:00:00    금 NaN  459.0  387.0  17.0  55.0  0.0   0.0  121.0   
2   2021-01-02 00:00:00    토 NaN  596.0  497.0  14.0  85.0  2.0   0.0  228.0   
3   2021-01-03 00:00:00    일 NaN  421.0  360.0  15.0  46.0  2.0   0.0  143.0   
4   2021-01-04 00:00:00    월 NaN   74.0   66.0   3.0   5.0  0.0   0.0   87.0   
5   2021-01-05 00:00:00    화 NaN   59.0   49.0   4.0   6.0  0.0   0.0   81.0   
..                  ...  ...  ..    ...    ...   ...   ...  ...   ...    ...   
28  2021-12-28 00:00:00    화 NaN  127.0  104.0   9.0  14.0  1.0   0.0  154.0   
29  2021-12-29 00:00:00    수 NaN  208.0  192.0   1.0  15.0  1.0   0.0  142.0   
30  2021-12-30 00:00:00    목 NaN  231.0  190.0  10.0  31.0  0.0   0.0  149.0   
31  2021-12-31 00:00:00    금 NaN   99.0   75.0  15.0   9.0  5.0   0.0   90.0   
32                  NaN  NaN NaN    NaN    NaN   NaN   NaN  NaN   NaN    NaN   

       총계  
1   580.0  
2   824.0  
3   564.0  
4   161.0  
5   140.0  
..    ...  
28  281.0  
29  350.0  
30  380.0  
31  189.0  
32    NaN  

[366 rows x 11 columns]

마지막 행을 삭제합니다.

In [71]:
seoulland_admission_2021=seoulland_admission_2021.iloc[:-1]

## 2022

10월까지 밖에 없습니다.

In [44]:
seoulland_admission_2022=pd.DataFrame([])
for i in range(0,10):
    seoulland_admission_2022=pd.concat([seoulland_admission_2022,pd.read_excel('Data/서울대공원 입장객 정보_2022년(1월-10월).xlsx',header=[4],engine='openpyxl',sheet_name=i)],axis=0)
    print(seoulland_admission_2022.shape)

(33, 25)
(63, 25)
(95, 25)
(126, 25)
(159, 25)
(191, 25)
(223, 25)
(255, 25)
(286, 25)
(318, 25)


다 맞습니다.

In [45]:
#컬럼 선택
seoulland_admission_2022=seoulland_admission_2022.iloc[:,[0,1,2,7,8,9,10,11,12,13,14]]

#컬럼명 변경
seoulland_admission_2022.columns=seoulland_admission_2018.columns

#불필요한 행 삭제
seoulland_admission_2022=seoulland_admission_2022.loc[~seoulland_admission_2022['날짜'].isin(['월 총계','월계']),:]

In [46]:
seoulland_admission_2022

날짜 요일  날씨     유료소계       어른    청소년     어린이   외국인  \
1   2022-01-01 00:00:00  토 NaN    790.0    636.0   38.0   116.0   2.0   
2   2022-01-02 00:00:00  일 NaN    749.0    612.0   37.0   100.0   0.0   
3   2022-01-03 00:00:00  월 NaN    188.0    159.0    3.0    26.0   0.0   
4   2022-01-04 00:00:00  화 NaN    175.0    142.0    7.0    26.0   0.0   
5   2022-01-05 00:00:00  수 NaN    166.0    132.0    6.0    28.0   0.0   
..                  ... ..  ..      ...      ...    ...     ...   ...   
27  2022-10-27 00:00:00  목 NaN   2263.0   1579.0  387.0   297.0   4.0   
28  2022-10-28 00:00:00  금 NaN   3334.0   2038.0  928.0   368.0   2.0   
29  2022-10-29 00:00:00  토 NaN  12404.0  10377.0  252.0  1775.0  59.0   
30  2022-10-30 00:00:00  일 NaN  25551.0  20898.0  522.0  4131.0   0.0   
31  2022-10-31 00:00:00  월 NaN   2182.0   1737.0  206.0   239.0   7.0   

       단체입장    무료소계       총계  
1       0.0   179.0    969.0  
2       0.0   237.0    986.0  
3       0.0   106.0    294.0  
4       0.0   198.0    373.0  
5       0.0   163.0    329.0  
..      ...     ...      ...  
27    700.0  1161.0   3424.0  
28    866.0   980.0   4314.0  
29    459.0  1993.0  14397.0  
30  12475.0  1331.0  26882.0  
31    181.0   846.0   3028.0  

[308 rows x 11 columns]

In [47]:
31+28+31+30+31+30+31+31+30+31

304

In [48]:
seoulland_admission_2022.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 308 entries, 1 to 31
Data columns (total 11 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   날짜      304 non-null    object 
 1   요일      304 non-null    object 
 2   날씨      0 non-null      float64
 3   유료소계    276 non-null    float64
 4   어른      304 non-null    float64
 5   청소년     304 non-null    float64
 6   어린이     304 non-null    float64
 7   외국인     303 non-null    float64
 8   단체입장    300 non-null    float64
 9   무료소계    304 non-null    float64
 10  총계      276 non-null    float64
dtypes: float64(9), object(2)
memory usage: 28.9+ KB


날짜가 NaN이 있는 것 같습니다.

In [49]:
seoulland_admission_2022[seoulland_admission_2022['날짜'].isna()]

날짜   요일  날씨  유료소계  어른  청소년  어린이  외국인  단체입장  무료소계  총계
32  NaN  NaN NaN   NaN NaN  NaN  NaN  NaN   NaN   NaN NaN
29  NaN  NaN NaN   NaN NaN  NaN  NaN  NaN   NaN   NaN NaN
32  NaN  NaN NaN   NaN NaN  NaN  NaN  NaN   NaN   NaN NaN
31  NaN  NaN NaN   NaN NaN  NaN  NaN  NaN   NaN   NaN NaN

In [52]:
seoulland_admission_2022=seoulland_admission_2022[~seoulland_admission_2022['날짜'].isna()]
seoulland_admission_2022

날짜 요일  날씨     유료소계       어른    청소년     어린이   외국인  \
1   2022-01-01 00:00:00  토 NaN    790.0    636.0   38.0   116.0   2.0   
2   2022-01-02 00:00:00  일 NaN    749.0    612.0   37.0   100.0   0.0   
3   2022-01-03 00:00:00  월 NaN    188.0    159.0    3.0    26.0   0.0   
4   2022-01-04 00:00:00  화 NaN    175.0    142.0    7.0    26.0   0.0   
5   2022-01-05 00:00:00  수 NaN    166.0    132.0    6.0    28.0   0.0   
..                  ... ..  ..      ...      ...    ...     ...   ...   
27  2022-10-27 00:00:00  목 NaN   2263.0   1579.0  387.0   297.0   4.0   
28  2022-10-28 00:00:00  금 NaN   3334.0   2038.0  928.0   368.0   2.0   
29  2022-10-29 00:00:00  토 NaN  12404.0  10377.0  252.0  1775.0  59.0   
30  2022-10-30 00:00:00  일 NaN  25551.0  20898.0  522.0  4131.0   0.0   
31  2022-10-31 00:00:00  월 NaN   2182.0   1737.0  206.0   239.0   7.0   

       단체입장    무료소계       총계  
1       0.0   179.0    969.0  
2       0.0   237.0    986.0  
3       0.0   106.0    294.0  
4       0.0   198.0    373.0  
5       0.0   163.0    329.0  
..      ...     ...      ...  
27    700.0  1161.0   3424.0  
28    866.0   980.0   4314.0  
29    459.0  1993.0  14397.0  
30  12475.0  1331.0  26882.0  
31    181.0   846.0   3028.0  

[304 rows x 11 columns]

## 모두 합치기(2018~2022)

In [73]:
seoulland_admission_2018

날짜 요일     날씨  유료소계    어른  청소년  어린이  외국인  단체입장  무료소계    총계
0   2018-01-01 00:00:00  월     맑음  1683  1375   99  209   40     0   619  2302
1   2018-01-02 00:00:00  화  구름 조금   401   342   15   44   22   119   578   979
2   2018-01-03 00:00:00  수  구름 조금   191   153   22   16    4     0   502   693
3   2018-01-04 00:00:00  목     흐림   163   132    9   22   14     1   488   651
4   2018-01-05 00:00:00  금  구름 조금   188   145   12   31    8     0   567   755
..                  ... ..    ...   ...   ...  ...  ...  ...   ...   ...   ...
26  2018-12-27 00:00:00  목     맑음  3498  2960   33  505   10  3422   418  3916
27  2018-12-28 00:00:00  금     맑음    70    59    7    4   14     0   258   328
28  2018-12-29 00:00:00  토     맑음   235   195   13   27   27     0   473   708
29  2018-12-30 00:00:00  일     맑음   445   368   21   56   37     0   436   881
30  2018-12-31 00:00:00  월     맑음   380   333   22   25   15     0   427   807

[365 rows x 11 columns]

In [66]:
seoulland_admission_2022

날짜 요일  날씨     유료소계       어른    청소년     어린이   외국인  \
1   2022-01-01 00:00:00  토 NaN    790.0    636.0   38.0   116.0   2.0   
2   2022-01-02 00:00:00  일 NaN    749.0    612.0   37.0   100.0   0.0   
3   2022-01-03 00:00:00  월 NaN    188.0    159.0    3.0    26.0   0.0   
4   2022-01-04 00:00:00  화 NaN    175.0    142.0    7.0    26.0   0.0   
5   2022-01-05 00:00:00  수 NaN    166.0    132.0    6.0    28.0   0.0   
..                  ... ..  ..      ...      ...    ...     ...   ...   
27  2022-10-27 00:00:00  목 NaN   2263.0   1579.0  387.0   297.0   4.0   
28  2022-10-28 00:00:00  금 NaN   3334.0   2038.0  928.0   368.0   2.0   
29  2022-10-29 00:00:00  토 NaN  12404.0  10377.0  252.0  1775.0  59.0   
30  2022-10-30 00:00:00  일 NaN  25551.0  20898.0  522.0  4131.0   0.0   
31  2022-10-31 00:00:00  월 NaN   2182.0   1737.0  206.0   239.0   7.0   

       단체입장    무료소계       총계  
1       0.0   179.0    969.0  
2       0.0   237.0    986.0  
3       0.0   106.0    294.0  
4       0.0   198.0    373.0  
5       0.0   163.0    329.0  
..      ...     ...      ...  
27    700.0  1161.0   3424.0  
28    866.0   980.0   4314.0  
29    459.0  1993.0  14397.0  
30  12475.0  1331.0  26882.0  
31    181.0   846.0   3028.0  

[304 rows x 11 columns]

In [74]:
seoulland_admission_18_22=pd.concat([seoulland_admission_2018,seoulland_admission_2019,seoulland_admission_2020,seoulland_admission_2021,seoulland_admission_2022])

In [75]:
seoulland_admission_18_22.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1765 entries, 0 to 31
Data columns (total 11 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   날짜      1765 non-null   object 
 1   요일      1765 non-null   object 
 2   날씨      806 non-null    object 
 3   유료소계    1736 non-null   float64
 4   어른      1764 non-null   float64
 5   청소년     1764 non-null   float64
 6   어린이     1764 non-null   float64
 7   외국인     1763 non-null   float64
 8   단체입장    1759 non-null   float64
 9   무료소계    1764 non-null   float64
 10  총계      1736 non-null   float64
dtypes: float64(8), object(3)
memory usage: 165.5+ KB


이렇게 5년치 데이터를 모두 모았습니다.

In [76]:
seoulland_admission_18_22

날짜 요일     날씨     유료소계       어른    청소년     어린이   외국인  \
0   2018-01-01 00:00:00  월     맑음   1683.0   1375.0   99.0   209.0  40.0   
1   2018-01-02 00:00:00  화  구름 조금    401.0    342.0   15.0    44.0  22.0   
2   2018-01-03 00:00:00  수  구름 조금    191.0    153.0   22.0    16.0   4.0   
3   2018-01-04 00:00:00  목     흐림    163.0    132.0    9.0    22.0  14.0   
4   2018-01-05 00:00:00  금  구름 조금    188.0    145.0   12.0    31.0   8.0   
..                  ... ..    ...      ...      ...    ...     ...   ...   
27  2022-10-27 00:00:00  목    NaN   2263.0   1579.0  387.0   297.0   4.0   
28  2022-10-28 00:00:00  금    NaN   3334.0   2038.0  928.0   368.0   2.0   
29  2022-10-29 00:00:00  토    NaN  12404.0  10377.0  252.0  1775.0  59.0   
30  2022-10-30 00:00:00  일    NaN  25551.0  20898.0  522.0  4131.0   0.0   
31  2022-10-31 00:00:00  월    NaN   2182.0   1737.0  206.0   239.0   7.0   

       단체입장    무료소계       총계  
0       0.0   619.0   2302.0  
1     119.0   578.0    979.0  
2       0.0   502.0    693.0  
3       1.0   488.0    651.0  
4       0.0   567.0    755.0  
..      ...     ...      ...  
27    700.0  1161.0   3424.0  
28    866.0   980.0   4314.0  
29    459.0  1993.0  14397.0  
30  12475.0  1331.0  26882.0  
31    181.0   846.0   3028.0  

[1765 rows x 11 columns]

날짜순으로 정렬하고 인덱스를 리셋하겠습니다.

날짜순 정렬하려면 우선 날짜를 datetime으로 변경해줍니다.

In [77]:
seoulland_admission_18_22['날짜']=seoulland_admission_18_22['날짜'].apply(lambda x:pd.to_datetime(x,infer_datetime_format=True))

In [78]:
seoulland_admission_18_22.sort_values(by='날짜',inplace=True)

In [79]:
seoulland_admission_18_22

날짜 요일     날씨     유료소계       어른    청소년     어린이   외국인     단체입장  \
0  2018-01-01  월     맑음   1683.0   1375.0   99.0   209.0  40.0      0.0   
1  2018-01-02  화  구름 조금    401.0    342.0   15.0    44.0  22.0    119.0   
2  2018-01-03  수  구름 조금    191.0    153.0   22.0    16.0   4.0      0.0   
3  2018-01-04  목     흐림    163.0    132.0    9.0    22.0  14.0      1.0   
4  2018-01-05  금  구름 조금    188.0    145.0   12.0    31.0   8.0      0.0   
..        ... ..    ...      ...      ...    ...     ...   ...      ...   
27 2022-10-27  목    NaN   2263.0   1579.0  387.0   297.0   4.0    700.0   
28 2022-10-28  금    NaN   3334.0   2038.0  928.0   368.0   2.0    866.0   
29 2022-10-29  토    NaN  12404.0  10377.0  252.0  1775.0  59.0    459.0   
30 2022-10-30  일    NaN  25551.0  20898.0  522.0  4131.0   0.0  12475.0   
31 2022-10-31  월    NaN   2182.0   1737.0  206.0   239.0   7.0    181.0   

      무료소계       총계  
0    619.0   2302.0  
1    578.0    979.0  
2    502.0    693.0  
3    488.0    651.0  
4    567.0    755.0  
..     ...      ...  
27  1161.0   3424.0  
28   980.0   4314.0  
29  1993.0  14397.0  
30  1331.0  26882.0  
31   846.0   3028.0  

[1765 rows x 11 columns]

이제 인덱스를 리셋합니다

In [80]:
seoulland_admission_18_22.reset_index(drop=True,inplace=True)

In [81]:
seoulland_admission_18_22

날짜 요일     날씨     유료소계       어른    청소년     어린이   외국인     단체입장  \
0    2018-01-01  월     맑음   1683.0   1375.0   99.0   209.0  40.0      0.0   
1    2018-01-02  화  구름 조금    401.0    342.0   15.0    44.0  22.0    119.0   
2    2018-01-03  수  구름 조금    191.0    153.0   22.0    16.0   4.0      0.0   
3    2018-01-04  목     흐림    163.0    132.0    9.0    22.0  14.0      1.0   
4    2018-01-05  금  구름 조금    188.0    145.0   12.0    31.0   8.0      0.0   
...         ... ..    ...      ...      ...    ...     ...   ...      ...   
1760 2022-10-27  목    NaN   2263.0   1579.0  387.0   297.0   4.0    700.0   
1761 2022-10-28  금    NaN   3334.0   2038.0  928.0   368.0   2.0    866.0   
1762 2022-10-29  토    NaN  12404.0  10377.0  252.0  1775.0  59.0    459.0   
1763 2022-10-30  일    NaN  25551.0  20898.0  522.0  4131.0   0.0  12475.0   
1764 2022-10-31  월    NaN   2182.0   1737.0  206.0   239.0   7.0    181.0   

        무료소계       총계  
0      619.0   2302.0  
1      578.0    979.0  
2      502.0    693.0  
3      488.0    651.0  
4      567.0    755.0  
...      ...      ...  
1760  1161.0   3424.0  
1761   980.0   4314.0  
1762  1993.0  14397.0  
1763  1331.0  26882.0  
1764   846.0   3028.0  

[1765 rows x 11 columns]

## 데이터 저장

In [122]:
seoulland_admission_18_22.to_csv('Data/seoulland_admission_18_22.csv',index=False)